### PR TITLE
Fixed the iPython and SSL with Tornado issues using GPU.

### LINKED ISSUE #86: Couldn't build the floydhub/dl-docker:gpu image
when running 
`docker build -t floydhub/dl-docker:gpu -f Dockerfile.gpu .`
the output in the terminal is : 

```
Collecting wcwidth (from prompt-toolkit<2.0.0,>=1.0.4->ipython)
  Downloading https://files.pythonhosted.org/packages/7e/9f/526a6947247599b084ee5232e4f9190a38f398d7300d866af3ab571a5bfe/wcwidth-0.1.7-py2.py3-none-any.whl
matplotlib 1.3.1 requires tornado, which is not installed.
Installing collected packages: scandir, pathlib2, pickleshare, ipython-genutils, traitlets, backports.shutil-get-terminal-size, pygments, wcwidth, prompt-toolkit, ipython
  Running setup.py install for scandir: started
    Running setup.py install for scandir: finished with status 'done'
  Found existing installation: ipython 1.2.1
Cannot uninstall 'ipython'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
The command '/bin/sh -c pip --no-cache-dir install --upgrade ipython && 	pip --no-cache-dir install 		Cython 		ipykernel 		jupyter 		path.py 		Pillow 		pygments 		six 		sphinx 		wheel 		zmq 		&& 	python -m ipykernel.kernelspec' returned a non-zero code: 1
```
i am running docker-ce, my nvidia driver is 390.48 and installed nvidia-docker2

### PR DESCRIPTION
**Pull request by huangyz0918**
Closed #86 #84

### CODE REVIEW

File: Dockerfile.gpu [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM nvidia/cuda:8.0-cudnn5-devel-ubuntu14.04
+FROM nvidia/cuda:9.2-cudnn7-devel-ubuntu16.04
```
Comment (huangyz0918) [inline Dockerfile.gpu]: This is fix for Caffe.

File: Dockerfile.gpu [hunk 2]
```diff
@@ -10,6 +10,16 @@ ARG LASAGNE_VERSION=v0.1
 ARG TORCH_VERSION=latest
 ARG CAFFE_VERSION=master
 
+ARG PYTHON_BASEDEPS="build-essential python-pip"
```
Comment (huangyz0918) [inline Dockerfile.gpu:13]: This is a fix for SSL.

File: Dockerfile.gpu [hunk 3]
```diff
@@ -106,15 +129,20 @@ RUN apt-get update && apt-get install -y \
 		python-nose \
 		python-h5py \
 		python-skimage \
-		python-matplotlib \
 		python-pandas \
 		python-sklearn \
-		python-sympy \
-		&& \
+		python-sympy
```
Comment (huangyz0918) [inline Dockerfile.gpu:134]: This is a fix for ipython

File: Dockerfile.gpu [hunk 4]
```diff
@@ -155,7 +183,9 @@ RUN apt-get update && apt-get install -y \
 	rm -rf /var/lib/apt/lists/*
 
 # Install Caffe
-RUN git clone -b ${CAFFE_VERSION} --depth 1 https://github.com/BVLC/caffe.git /root/caffe && \
+# Use video friendly Caffe instead.
```
Comment (huangyz0918) [inline Dockerfile.gpu:186]: use video friendly caffe.

File: Dockerfile.gpu (hunk 2 above)
Comment (OneCodeMonkey) [reply Dockerfile.gpu]: 不顶用啊，老哥。。。

Comment (OneCodeMonkey) [reply Dockerfile.gpu]: 我用的 cpu 模式，加了这行还是有这个错：`Tornado requires an up-to-date SSL module. This means Python 2.7.9+ or 3.4+ (although some distributions have backported the necessary changes to older versions).`

Comment (OneCodeMonkey) [reply Dockerfile.gpu]: `ARG PYTHON_BASEDEPS="build-essential python-pip"
ARG PYTHON_BUILDDEPS="libbz2-dev \
   libc6-dev \
   libgdbm-dev \
   libncursesw5-dev \
   libreadline-gplv2-dev \
   libsqlite3-dev \
   libssl-dev \
   tk-dev"


ARG PYTHON_TARFILE="Python-2.7.9.tgz"
ARG PYTHON_TARHOST="https://www.python.org/ftp/python/2.7.9"
ARG PYTHON_SRCDIR="Python-2.7.9"

RUN apt-get update
RUN apt-get install -y ${PYTHON_BASEDEPS} ${PYTHON_BUILDDEPS}

RUN wget "${PYTHON_TARHOST}/${PYTHON_TARFILE}"
RUN tar xvf ${PYTHON_TARFILE}
 
RUN cd ${PYTHON_SRCDIR} && \
   ./configure && \
   make && \
   make install`

**Pull request closed without merging**